fuzzing/fuzzer_test.go

```diff
@@ -396,6 +396,29 @@ func TestDeploymentsInternalLibrary(t *testing.T) {
 	})
 }
 
+// TestDeploymentsWithPredeploy runs a test to ensure that predeployed contracts are instantiated correctly.
+func TestDeploymentsWithPredeploy(t *testing.T) {
+	runFuzzerTest(t, &fuzzerSolcFileTest{
+		filePath: "testdata/contracts/deployments/predeploy_contract.sol",
+		configUpdates: func(config *config.ProjectConfig) {
+			config.Fuzzing.TargetContracts = []string{"TestContract"}
+			config.Fuzzing.TestLimit = 1000 // this test should expose a failure immediately
+			config.Fuzzing.Testing.PropertyTesting.Enabled = false
+			config.Fuzzing.Testing.OptimizationTesting.Enabled = false
+			config.Fuzzing.PredeployedContracts = map[string]string{"PredeployContract": "0x1234"}
+		},
+		method: func(f *fuzzerTestContext) {
+			// Start the fuzzer
+			err := f.fuzzer.Start()
+			assert.NoError(t, err)
+
+			// Check for any failed tests and verify coverage was captured
+			assertFailedTestsExpected(f, true)
+			assertCorpusCallSequencesCollected(f, true)
```

Comment:
I would expect this to fail without `TestAllContracts` being enabled. Are `PredeployedContracts` treated differently when they're not in the `TargetContracts` and `TestAllContracts = false`

Reply:
yes, a predeployed contract is not a target contract. when the contract is deployed on-chain and the event is emitted and captured by the fuzzer worker, it will only add the predeployed contract's methods if `TestAllContracts` is true. 

However, there is an interesting caveat with this PR which is you can predeploy a contract and also have it as part of `TargetContracts`. So, you can have a determinstic deployment of contract `A` which is not a target contract but then also deploy `A` as a target contract for fuzzing. Point being: a function in a predeployed contract is only called if (1) it's also a target contract or (2) `TestAllContracts` is true.

Reply:
You will notice in the solidity test file for this PR, that I explicitly call `triggerFailure` in the `TestContract`'s function. If that isn't there, there is no way to invoke the `triggerFailure` function. However, if we set `TestAllContracts` to true, then you wouldn't have to explicitly call it. 

P.S. I tried running it with `TestAllContracts` as `true` and removing the explicit invocation of `triggerFailure` and it worked.